bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: Return Kafka admin server URL in kas-fleet-manager Kafka Resource.

## Description
Return Admin Server URL as it is currently being manually constructed
[MGDSTRM-8163](https://issues.redhat.com/browse/MGDSTRM-8163)

Note:
 The Admin Server URL will only be returned when a kafka is in a `ready` status.

## Verification Steps
- Create a cluster on [OpenShift](https://console.redhat.com/openshift/) or with [CRC](https://crc.dev/crc/)
- Deploy kas-fleet-manager

To get the kafka information,you may run: 
`curl -v -XGET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/kafkas/<kafkaID> | jq`

Once `"status": "ready",` your output should look like this:
```json
{
"id": "ca6ao5f13honsptlr6mg",
"kind": "Kafka",
"href": "/api/kafkas_mgmt/v1/kafkas/ca6ao5f13honsptlr6mg",
"status": "ready",
"cloud_provider": "aws",
"multi_az": true,
"region": "us-east-1",
"owner": "egallina_kafka_service",
"name": "serviceapi",
"bootstrap_server_host": "[serviceapi-ca-ao-f--honsptlr-mg.kas.egallina-kfm.qgrd.s1.devshift.org:443](http://serviceapi-ca-ao-f--honsptlr-mg.kas.egallina-kfm.qgrd.s1.devshift.org:443/)",
"admin_api_server_url": "http://admin-server-serviceapi-ca-ao-f--honsptlr-mg.kas.egallina-kfm.qgrd.s1.devshift.org/",
"created_at": "2022-05-24T10:57:41.45165+01:00",
"updated_at": "2022-05-24T11:05:44.178069+01:00",
"version": "3.0.1",
"instance_type": "standard",
"instance_type_name": "Standard",
"reauthentication_enabled": true,
"kafka_storage_size": "1000Gi",
"browser_url": "http://localhost:8080/ca6ao5f13honsptlr6mg/dashboard",
"size_id": "x1",
"ingress_throughput_per_sec": "50Mi",
"egress_throughput_per_sec": "100Mi",
"total_max_connections": 3000,
"max_partitions": 1500,
"max_data_retention_period": "P14D",
"max_connection_attempts_per_sec": 100
}
```

Note:
This is the URL we have returned:
```diff
+ "admin_api_server_url": "http://admin-server-serviceapi-ca-ao-f--honsptlr-mg.kas.egallina-kfm.qgrd.s1.devshift.org/",
```

#### Check the Database:
You may also check that the url has been added to the database in the kafka_requests table as `admin_api_server_url`
- To access the database, run `make db/login`.
- To pull the contents of admin_api_server_url, run `select admin_api_server_url from kafka_requests;` 
Your output should look as follows: 

```
                                   admin_api_server_url                                    
-------------------------------------------------------------------------------------------
 http://admin-server-serviceapi-ca-afhv--hol-rnheg-g.kas.egallina-kfm.qgrd.s1.devshift.org
(1 row)
```

## Checklist (Definition of Done)
